packages/react-charting/src/components/LineChart/LineChart.base.tsx

```diff
@@ -223,11 +223,26 @@ export class LineChartBase extends React.Component<ILineChartProps, ILineChartSt
   }
 
   public render(): JSX.Element {
-    const { tickValues, tickFormat, eventAnnotationProps, legendProps, data } = this.props;
+    // eslint-disable-next-line @fluentui/max-len
+    const {
+      tickValues,
+      tickFormat,
+      eventAnnotationProps,
+      legendProps,
+      data,
+      disableSortByXValues = false,
+    } = this.props;
     this._points = this._injectIndexPropertyInLineChartData(data.lineChartData);
 
     const isXAxisDateType = getXAxisType(this._points);
     let points = this._points;
+
+    if (disableSortByXValues === false) {
```

Comment:
The sorting needs to happen in createLines also. Basically wherever this._points is getting reevaluated.

Reply:
Will make these changes...

Reply:
So... wherever this._points will be used... by default... points[i].data must be sorted... So can we do this in the constructor itself??

Reply:
Done